Source/NSBrowser.m

```diff
@@ -3205,58 +3242,103 @@ - (void) _remapColumnSubviews: (BOOL)fromFirst
   if (fromFirst)
     {
       for (i = _firstVisibleColumn; i <= _lastVisibleColumn; i++)
-        {
-          bc = [_browserColumns objectAtIndex: i];
-          sc = [bc columnScrollView];
-          [self addSubview: sc];
-
-          if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
-            {
-              [_window makeFirstResponder: firstResponder];
-              setFirstResponder = YES;
-            }
-        }
+	{
+	  bc = [_browserColumns objectAtIndex: i];
+	  sc = [bc columnScrollView];
+	  [self addSubview: sc];
+
+	  if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
+	    {
+	      [_window makeFirstResponder: firstResponder];
+	      setFirstResponder = YES;
+	    }
+	}
 
       if (firstResponder && setFirstResponder == NO)
-        {
-          [_window makeFirstResponder:
-                     [[_browserColumns objectAtIndex: _firstVisibleColumn]
-                       columnMatrix]];
-        }
+	{
+	  [_window makeFirstResponder:
+		     [[_browserColumns objectAtIndex: _firstVisibleColumn]
+		       columnMatrix]];
+	}
     }
   else
     {
       for (i = _lastVisibleColumn; i >= _firstVisibleColumn; i--)
-        {
-          bc = [_browserColumns objectAtIndex: i];
-          sc = [bc columnScrollView];
-          [self addSubview: sc];
-
-          if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
-            {
-              [_window makeFirstResponder: firstResponder];
-              setFirstResponder = YES;
-            }
-        }
+	{
+	  bc = [_browserColumns objectAtIndex: i];
+	  sc = [bc columnScrollView];
+	  [self addSubview: sc];
+
+	  if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
+	    {
+	      [_window makeFirstResponder: firstResponder];
+	      setFirstResponder = YES;
+	    }
+	}
 
       if (firstResponder && setFirstResponder == NO)
-        {
-          [_window makeFirstResponder:
-                     [[_browserColumns objectAtIndex: _lastVisibleColumn]
-                       columnMatrix]];
-        }
+	{
+	  [_window makeFirstResponder:
+		     [[_browserColumns objectAtIndex: _lastVisibleColumn]
+		       columnMatrix]];
+	}
     }
 }
 
+- (id) _itemForColumn: (NSInteger)column
+{
+  id item = nil;
+
+  if (column == 0)
+    {
+      item = [_browserDelegate rootItemForBrowser: self];
+    }
+  else
+    {
+      NSInteger col = column > 0 ? column - 1 : column;
```

Comment:
This line seems strange. Do we have negative columns? If not the value could either be 0 or bigger than zero and the zero case is already handled in the previous `if` case, so there is no need to be checked here.

Reply:
We don't have negative columns.   The idea here is that if column is 0, then col will be zero and so we get the first column.   If it is greater than zero we get the previous column.   I will take a look and optimize this code.

Reply:
I have simplified this code as discussed.